### PR TITLE
Make it possible to test locally without needing minikube

### PR DESCRIPTION
**Pull request by yuvipanda**
+ minor URL changes to installing minikube
+ expand minikube instructions to be self-sufficient

### CODE REVIEW

**Comment by yuvipanda** [comment]
Someone with an OS X laptop should test this. The local test stuff should work for windows too.

**Comment by yuvipanda** [comment]
Apparently the local test stuff does not work on Windows, but might under WSL :) @choldgraf is testing!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,26 +2,70 @@
...
+   ```
+
+2. Run it!
+
+   ```bash
+   python3 -m binderhub.app -f testing/localonly/binderhub_config.py
+   ```
 
-        python3 -m binderhub -f testing/binderhub_config.py
+3. You can now access it locally at https://localhost:8585
```
Comment (willingc) [inline CONTRIBUTING.md]: Running locally works for view css/html but there is an error thrown when there is not a valid GH repo on start.
```
$ python3 -m binderhub.app -f testing/localonly/binderhub_config.py --debug
[BinderHub] Looking for testing/localonly/binderhub_config in /Users/carol/code/jhub/binderhub
[BinderHub] Loaded config file: /Users/carol/code/jhub/binderhub/testing/localonly/binderhub_config.py
[I 171007 00:00:22 app:261] BinderHub starting on port 8585
[I 171007 00:00:36 web:2063] 304 GET / (::1) 18.64ms
[E 171007 00:00:40 builder:120] Failed to get provider for gh:/master
    Traceback (most recent call last):
      File "/Users/carol/code/jhub/binderhub/binderhub/builder.py", line 118, in get
        provider = self.get_provider(provider_prefix, spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/base.py", line 19, in get_provider
        config=self.settings['traitlets_config'], spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/repoproviders.py", line 107, in __init__
        raise ValueError(msg)
    ValueError: Spec is not of the form "user/repo/ref", provided: "/master".
[I 171007 00:00:40 web:2063] 200 GET /build/gh//master (::1) 2.27ms
```

**Comment by yuvipanda** [comment]
Tyvm for testing @willingc. Yeah, that error could be nicer in the ux - but
is the expected result of entering an invalid gh reference. Hopefully
making this easier to set up makes it easier for more people to contribute
so we can fixup the ux sooner :)

If someone doesn't merge this before I do I shall do so when I am on a
laptop next.

On Oct 6, 2017 9:06 PM, "Carol Willing" <notifications@github.com> wrote:

*@willingc* commented on this pull request.

This works for seeing the css/html locally on MacOS. There is an error on
startup when a valid GH location is not available.
------------------------------

In CONTRIBUTING.md
<https://github.com/jupyterhub/binderhub/pull/175#discussion_r143321980>:

>
-        python3 -m binderhub -f testing/binderhub_config.py
+3. You can now access it locally at https://localhost:8585

Running locally works for view css/html but there is an error thrown when
there is not a valid GH repo on start.

$ python3 -m binderhub.app -f testing/localonly/binderhub_config.py --debug
[BinderHub] Looking for testing/localonly/binderhub_config in
/Users/carol/code/jhub/binderhub
[BinderHub] Loaded config file:
/Users/carol/code/jhub/binderhub/testing/localonly/binderhub_config.py
[I 171007 00:00:22 app:261] BinderHub starting on port 8585
[I 171007 00:00:36 web:2063] 304 GET / (::1) 18.64ms
[E 171007 00:00:40 builder:120] Failed to get provider for gh:/master
    Traceback (most recent call last):
      File "/Users/carol/code/jhub/binderhub/binderhub/builder.py",
line 118, in get
        provider = self.get_provider(provider_prefix, spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/base.py", line
19, in get_provider
        config=self.settings['traitlets_config'], spec=spec)
      File "/Users/carol/code/jhub/binderhub/binderhub/repoproviders.py",
line 107, in __init__
        raise ValueError(msg)
    ValueError: Spec is not of the form "user/repo/ref", provided: "/master".
[I 171007 00:00:40 web:2063] 200 GET /build/gh//master (::1) 2.27ms

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/jupyterhub/binderhub/pull/175#pullrequestreview-67824659>,
or mute the thread

**Comment by willingc** [comment]
I'm going to merge.

> That error could be nicer in the ux - but
is the expected result of entering an invalid gh reference. 

@yuvipanda To clarify, the error happens before there is any interaction with the UI.

**Comment by willingc** [comment]
@yuvipanda I'm sorry. I just tried it again and the error happens on first interaction with the UI. 👍 to fixing over time.

**Pull request merged**